### DISCUSSION TITLE
Custom Filetype Detection

### DISCUSSION THREAD

**Discussion by hrkcz001** [nvim-tree/nvim-tree.lua#2233] [Q&A] [answered]
Is it possible to explicitly change icons for files that match a particular pattern? For example, I have changed the filetypes for C++ headers, but it doesn't work in Nvim Tree.
```
ft.add {
    pattern = {
        ['.*/include/.*'] = 'cpp',
    },
}
```
I'm looking for something like this.

**Comment by alex-courtis** [comment, marked as answer]
Yes! You can customise icons / colours for names and filetypes via https://github.com/nvim-tree/nvim-web-devicons
